python_tests/mpi_test.py

```diff
@@ -0,0 +1,61 @@
+# run it with
+# NVFUSER_DUMP=fusion_ir mpirun --allow-run-as-root -n 2 python mpi_nvfuser.py
+
+import torch
+import nvfuser
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from torch import Tensor
+from typing import Tuple
+
+from nvfuser import FusionDefinition, DataType
+
+import os
+
+rank = int(os.environ["OMPI_COMM_WORLD_LOCAL_RANK"])
+
+torch.cuda.set_device(rank)
+
+### Inputs
+inputs = [
+    torch.randn(2, 4, device="cuda")[rank:rank+1, ...],
+]
+
+class MultiDeviceModel(FusionDefinition):
+
+    def definition(self):
+        # dynamic shape isn't working? at least I'm getting asserts
```

Comment:
Indeed we only support concrete shapes for now. Sorry, we probably forgot to mention it.
This is not exactly a restriction from the implementation side, but rather a convention issue because: since all the ranks won't necessarily have concrete inputs, we need to agree to a way to communicate concrete shapes at execution startup. In the meantime, we only focus on concrete tensors.